cms/src/test/java/uk/nhs/digital/common/OrderedSearchDateDerivedDataFunctionTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.nhs.digital.common;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import org.apache.jackrabbit.value.BooleanValue;
+import org.apache.jackrabbit.value.DateValue;
+import org.apache.jackrabbit.value.LongValue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.jcr.Value;
+
+@RunWith(DataProviderRunner.class)
+public class OrderedSearchDateDerivedDataFunctionTest {
+
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd-MM-yyyy");
+    private static final String ORDERED_SEARCH_DATE = "orderedSearchDate";
+
+    private OrderedSearchDateDerivedDataFunction function = new OrderedSearchDateDerivedDataFunction();
+
+    @Test
+    public void testUpcoming() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("20-03-2020", false);
+
+        long expectedValue = -parseDate("20-03-2020").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testPublished() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("10-08-2010", true);
+
+        long expectedValue = parseDate("10-08-2010").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testNoAccessibleFlag() throws Exception {
+        Map<String, Value[]> result = testFunctionCompute("18-11-2013", null);
+
+        long expectedValue = parseDate("18-11-2013").getTimeInMillis();
+
+        assertThat("Result is as expected", result, hasEntry(ORDERED_SEARCH_DATE, new Value[]{new LongValue(expectedValue)}));
+    }
+
+    @Test
+    public void testNoDate() throws Exception {
+        assertNull(testFunctionCompute(null, true).get(ORDERED_SEARCH_DATE));
+
+        assertNull(testFunctionCompute(null, false).get(ORDERED_SEARCH_DATE));
+
+        assertNull(testFunctionCompute(null, null).get(ORDERED_SEARCH_DATE));
+    }
+
+    @Test
+    public void testOrder() throws Exception {
```

Comment:
The block of `addData` gives an impression of setting up expectations, whilst, in fact, it performs the actual test by calling `OrderedSearchDateDerivedDataFunction.compute`.

Continuing the spirit of making the tests easier to read by more clearly represent the setup-test-assertion blocks (the given/when/then) and given that earlier test methods consistently make an explicit call to `testFunctionCompute`, how about:
* repurposing the map to only store inputs (date and 'publicly accessible' flag),
* making the first operation on the stream a `.map` call to `testFunctionCompute` to produce outputs.